metadata-types/src/types/HasuraMetadataV2.ts

```diff
@@ -0,0 +1,828 @@
+}
+
+/**
+ * https://hasura.io/docs/1.0/graphql/manual/api-reference/schema-metadata-api/custom-types.html#objecttype
+ */
+export interface ObjectType {
+  /** Name of the Input object type */
+  name: GraphQLName
+  /** Description of the Input object type */
+  description?: string
+  /** Fields of the Input object type */
+  fields: InputObjectField[]
+  /**Relationships of the Object type to tables */
+  relationships?: CustomTypeObjectRelationship[]
+}
+
+/**
+ * https://hasura.io/docs/1.0/graphql/manual/api-reference/schema-metadata-api/custom-types.html#objectfield
+ */
+export interface ObjectField {
+  /** Name of the Input object type */
+  name: GraphQLName
+  /** Description of the Input object type */
+  description?: string
+  /** GraphQL type of the Input object type */
+  type: GraphQLType
+}
+
+/**
+ * https://hasura.io/docs/1.0/graphql/manual/api-reference/schema-metadata-api/custom-types.html#objectrelationship
+ */
+export interface CustomTypeObjectRelationship {
+  /** Name of the relationship, shouldn’t conflict with existing field names */
+  name: RelationshipName
+  /** Type of the relationship */
+  type: 'object' | 'array'
+  /** The table to which relationship is defined */
+  remote_table: TableName
+  /** Mapping of fields of object type to columns of remote table  */
+  field_mapping: {
+    [ObjectFieldName: string]: string
+  }
+}
+
+/**
+ * https://hasura.io/docs/1.0/graphql/manual/api-reference/schema-metadata-api/custom-types.html#scalartype
+ */
+export interface ScalarType {
+  /** Name of the Scalar type */
+  name: GraphQLName
+  /** Description of the Scalar type */
+  description?: string
+}
+
+/**
+ * https://hasura.io/docs/1.0/graphql/manual/api-reference/schema-metadata-api/custom-types.html#enumtype
+ */
+export interface EnumType {
+  /** Name of the Enum type */
+  name: GraphQLName
+  /** Description of the Enum type */
+  description?: string
+  /** Values of the Enum type */
+  values: EnumValue[]
+}
+
+/**
+ * https://hasura.io/docs/1.0/graphql/manual/api-reference/schema-metadata-api/custom-types.html#enumvalue
+ */
+export interface EnumValue {
+  /** Value of the Enum type */
+  value: GraphQLName
+  /** Description of the Enum value */
+  description?: string
+  /** If set to true, the enum value is marked as deprecated */
+  is_deprecated?: boolean
+}
+
+////////////////////////////////
+//  #endregion CUSTOM TYPES
+///////////////////////////////
+
+////////////////////////////////
+// #region ACTIONS
+///////////////////////////////
+
+// Note: Action Metadata type is slightly different than "create" type
+
+/**
+ *
+ * https://hasura.io/docs/1.0/graphql/manual/api-reference/schema-metadata-api/actions.html#args-syntax
+ */
+export interface Action {
+  /** Name of the action  */
+  name: ActionName
+  /** Definition of the action */
+  definition: ActionDefinition
+  /** Comment */
+  comment?: string
+  /** Permissions of the action */
+  permissions?: { role: string }
```

Comment:
Hi I was using the types in a project, I think this should be 
```tsx
    permissions?: { role: string }[]
```
This is to give multiple roles permissions for an action

Reply:
Ahh good catch, thank you. Will fix today. Need to get more sample `metadata.json` to put in the `tests` directory to catch this.

https://github.com/GavinRay97/graphql-engine/tree/feature/add-metadata-types-and-generator/metadata-types/src/tests